dbms/src/IO/HTTPCommon.cpp

```diff
@@ -196,7 +197,18 @@ PooledHTTPSessionPtr makePooledHTTPSession(const Poco::URI & uri, const Connecti
 std::istream * receiveResponse(
     Poco::Net::HTTPClientSession & session, const Poco::Net::HTTPRequest & request, Poco::Net::HTTPResponse & response)
 {
-    auto istr = &session.receiveResponse(response);
+    try
+    {
+        auto istr = &session.receiveResponse(response);
+    }
+    catch ( const Poco::Net::Exception & e )
+    {
+        log = &Logger::get("IOHTTP");
+        LOG_DEBUG(log, "reason: " << e.displayText());
+        &session->reset();
+        // Retry once
+        istr = &session.receiveResponse(request);
```

Comment:
I think, the logic of retry is better to make in loop.
Maybe, add parameter `Retries` in `Settings.h` and loop `Retries` time. If after `Retries` count still we have exceptions, so log the error, throw exception, etc.

Reply:
Good call.  I'm going to test this in our deployment before I dive into configuration, but should definitely be configurable.  I'll look around the codebase for other examples of retries and try to match them, unless a contributor beats me to it.